paper/paper.md

```diff
@@ -0,0 +1,84 @@
+---
+title: 'nbgrader: A Tool for Creating and Grading Assignments in the Jupyter Notebook'
+tags:
+- jupyter
+- jupyterhub
+- grading
+- autograding
+authors:
+- name: Project Jupyter
+- name: Douglas Blank
+  orcid: 0000-0003-3538-8829
+  affiliation: 7
+- name: David Bourgin
+  orcid: 0000-0003-1039-6195
+  affiliation: 1
+- name: Matthias Bussonnier
+  orcid: 
+  affiliation: 1
+- name: Jonathan Frederic
+  orcid: 0000-0003-4805-2216
+  affiliation: 2
+- name: Brian Granger
+  orcid: 
+  affiliation: 3
+- name: Jessica Hamrick
+  orcid: 0000-0002-3860-0429
+  affiliation: 4
+- name: Logan Page
+  orcid: 0000-0002-5799-8524
+  affiliation: 5
+- name: Fernando Pérez
+  orcid: 
+  affiliation: 1
+- name: Benjamin Ragan-Kelley
+  orcid: 0000-0002-1023-7082
+  affiliation: 6
+- name: Carol Willing
+  orcid: 0000-0002-9817-8485
+  affiliation: 3
+affiliations:
+- name: University of California, Berkeley
+  index: 1
+- name: Google Inc.
+  index: 2
+- name: Cal Poly, San Luis Obispo
+  index: 3
+- name: DeepMind
+  index: 4
+- name: University of Pretoria
+  index: 5
+- name: Simula Research Laboratory
+  index: 6
+- name: Bryn Mawr College
+  index: 7
+date: 2 June 2018
+bibliography: paper.bib
+---
+
+*Note: Authors on this paper are listed in alphabetical order.*
+
+# Summary
+
+nbgrader is a tool for creating and grading assignments in the Jupyter notebook [@kluyver2016jupyter]. nbgrader allows instructors to create a single, master copy of the assignment, from which the student version is generated—thus obviating the need to maintain two separate versions. nbgrader automatically grades submitted assignments by executing the notebooks and storing the results in a database. After auto-grading, instructors can provide partial credit or manually grade free-responses using the *formgrader* notebook extension. Finally, instructors can use nbgrader to leave personalized feedback for each student, including instructor comments as well as detailed error information.
+
+When used with JupyterHub [@JupyterHub], nbgrader provides additional workflow functionality that covers the entire grading process. After creating the assignment, instructors can release it to students, who can then fetch a copy of the assignment directly through the notebook server interface. Students can submit their completed version through the same interface, making it available for instructors to collect with one command. After that, instructors can use the auto-grading functionality as normal and may access the formgrader as a JupyterHub service.
```

Comment:
- [ ] (warning: extreme pedantry) `that covers` is mildly illogical here, because the added functionality covers those parts of the grading process not already covered without JupyterHub. As it's written, it says that the additional functionality is covers everything. I think you can fix this by changing `that covers` => `, covering"  or `and covers`
- [ ] `completed version` => `completed assignment`, just to cut down on overloading of `version` and `copy`, which I think are pretty much being used synonymously but I'm not totally sure at this point.
- [ ] Cleanly split the submission and grading in the last two sentences. "Students can submit their completed assignments through the same interface, making it available to instructors. Afterwards, instructors can collect the assignments with a single command and use the auto-grading functionality in the normal way. The formgrader can also be accessed as a JupyterHub service." 
- [ ] Will the average person reading this know what a JupyterHub service is?

Reply:
- [ ] `After creating the assignment` => `After creating an assignment`
- [ ] `for instructors to collect with one command`: it might be ambiguous what "collect" means here (this isn't an explicit part of other grading interfaces like gradeScope or Canvas, where "collection" is immediate). Perhaps it could be changed to something like `making it available to instructors with a single command` ? Alternatively, @suchow 's suggestion about reflowing the last two sentences would help resolve this by contrasting collection and autograding within the same sentence.